packages/labs/src/multi-select/multiSelect.tsx

```diff
@@ -0,0 +1,299 @@
+
+        if (openOnKeyDown) {
+            nextState = { ...nextState, isOpen: false };
+        }
+        if (resetOnSelect) {
+            nextState = { ...nextState, query: "" };
+        }
+
+        this.setState(nextState);
+    }
+
+    private deselectItem(index: number) {
+        this.setState({ selectedItems: this.state.selectedItems.filter((_, i) => i !== index) });
+    }
+
+    private handleTagRemove = (_item: string, index: number) => {
+        this.deselectItem(index);
+
+        Utils.safeInvoke(this.props.tagInputProps.onRemove, _item, index);
+    }
+
+    private handleItemSelect = (item: T, _event: React.SyntheticEvent<HTMLElement>) => {
+        // check whether the query string is in the item list
+        if (item !== undefined && this.state.activeItem !== undefined) {
+            if (!this.isItemSelected(item)) {
+                this.selectItem(item);
+            } else {
+                this.deselectItem(this.getSelectedItemIndex(item));
+            }
+        }
+        this.input.focus();
+
+        Utils.safeInvoke(this.props.onItemSelect, item, _event);
+    }
+
+    private handlePopoverInteraction = (nextOpenState: boolean) => requestAnimationFrame(() => {
+        // deferring to rAF to get properly updated activeElement
+        const { popoverProps = {} } = this.props;
+
+        if (this.input != null && this.input !== document.activeElement) {
+            // the input is no longer focused so we can close the popover
+            let nextState = { isOpen: false };
+
+            if (this.props.resetOnSelect) {
+                nextState = { ...nextState, query: "" };
+            }
+
+            this.setState(nextState);
+        } else if (!this.props.openOnKeyDown) {
+            // open the popover when focusing the tag input
+            this.setState({ isOpen: true });
+        }
+
+        Utils.safeInvoke(popoverProps.onInteraction, nextOpenState);
+    })
+
+    private handlePopoverDidOpen = () => {
+        const { popoverProps = {} } = this.props;
+
+        if (this.queryList != null) {
+            // scroll active item into view after popover transition completes and all dimensions are stable.
+            this.queryList.scrollActiveItemIntoView();
+        }
+
+        Utils.safeInvoke(popoverProps.popoverDidOpen);
+    }
+
+    private handleActiveItemChange = (activeItem: T) => {
+        this.setState({ activeItem });
+    }
+
+    private handleKeyDown = (event: React.KeyboardEvent<HTMLElement>) => {
+        if (event.which === Keys.ESCAPE
+            || event.which === Keys.TAB) {
+            let nextState = { isOpen: false };
+
+            if (this.props.resetOnSelect) {
+                nextState = { ...nextState, query: "" };
+            }
+
+            this.setState(nextState);
+        } else if (!(event.which === Keys.ARROW_LEFT
+           || event.which === Keys.ARROW_RIGHT
+           || event.which === Keys.BACKSPACE)) {
```

Comment:
_but actually_ `TagInput` calls `preventDefault()` when those keys are triggered... maybe you can just check `event.isDefaultPrevented` instead of redefining this logic?